Dockerfile

```diff
@@ -0,0 +1,9 @@
+FROM openjdk
+
+# Install libsodium
+RUN apt-get update && apt-get install -y \
+        build-essential \
+        software-properties-common
+
+COPY ./install-libsodium.sh .
+RUN ./install-libsodium.sh
```

Comment:
Looks like the PPA has not been updated since 2015!  In which case, my second suggestion.  This achieves the same end-aim (fewer layers in the final image, and less unwanted bits):

The `COPY`/`RUN` commands in the file can become:

```
ENV LIBSODIUM_VERSION=1.0.16

RUN mkdir /build && \
        curl "https://download.libsodium.org/libsodium/releases/libsodium-${LIBSODIUM_VERSION}.tar.gz" | \
                tar xz --directory /build --strip 1 && \
        cd /build && \
        ./configure && \
        make all check install && \
        rm -rf /build
```

and the script can be dropped.  This still leaves `build-essential` in the final image, which a ppa install would avoid, but apart from that it should be fairly minimal

Reply:
The reason for having a separate script was so that a user wanting to run the API locally could run the same install script used by the API container.  So I'd like to leave the script in the repo for that purpose (and just manually keep the script and the Dockerfile up to date)